src/dotty/tools/dotc/repl/CompilingInterpreter.scala

```diff
@@ -383,24 +387,54 @@ class CompilingInterpreter(out: PrintWriter, ictx: Context) extends Compiler wit
       names1 ++ names2
     }
 
+    /** Sets both System.{out,err} and Console.{out,err} to supplied
+     *  `os: OutputStream`
+     */
+    private def withOutput[T](os: ByteOutputStream)(op: ByteOutputStream => T) = {
+      val ps     = new PrintStream(os)
+      val oldOut = System.out
+      val oldErr = System.err
+      System.setOut(ps)
+      System.setErr(ps)
+
+      try {
+        Console.withOut(os)(Console.withErr(os)(op(os)))
+      } finally {
+        System.setOut(oldOut)
+        System.setErr(oldErr)
+      }
+    }
+
     /** load and run the code using reflection.
      *  @return  A pair consisting of the run's result as a string, and
```

Comment:
That comment is no longer accurate, the first element of the pair is no longer a string